### PR TITLE
NovelRT Engine plugins and library loading support.

### PR DESCRIPTION
**Pull request by RubyNova**
This is a misc. feature required for our future standardisation of engine backends for systems such as input, audio, graphics, etc.

It also includes a little cross-platform utility for library loading, to make runtime library loading a little easier. 😄

### CODE REVIEW

File: tests/NovelRT.Tests/CMakeLists.txt
Comment (capnkenny) [inline tests/NovelRT.Tests/CMakeLists.txt:68]: Wasn't there a fix we had  for this? Not sure if this is still necessary or not

**Review by capnkenny — COMMENTED** [review]
Overall looks good, just a couple questions in my mind that can wait for the ol 'cord 👍

File: include/NovelRT/Plugins/PluginKind.h
Comment (RubyNova) [reply include/NovelRT/Plugins/PluginKind.h:16]: it would more likely be `LanguageService` or `LanguageSystem`, but yes. :)

File: tests/NovelRT.Tests/CMakeLists.txt
Comment (RubyNova) [reply tests/NovelRT.Tests/CMakeLists.txt:68]: not afaik?

Comment (capnkenny) [reply tests/NovelRT.Tests/CMakeLists.txt:68]: I double checked and I don't see it either - must've been dreaming :^)

**Review by capnkenny — APPROVED** [review]
After reviewing at a normal time in the AM, I answered my own question. No further questions, and I think this looks good - LGTM (once you pull merge master and confirm it works again)

File: src/NovelRT/Graphics/RenderingService.cpp
Comment (FiniteReality) [inline src/NovelRT/Graphics/RenderingService.cpp:10]: What on earth is the indentation here doing?

File: tests/DoTest/DoTest.cpp
Comment (FiniteReality) [inline tests/DoTest/DoTest.cpp:4]: Bracing inconsistency?

File: samples/Simple/main.cpp
Comment (FiniteReality) [inline samples/Simple/main.cpp:33]: More brace inconsistencies?

**Review by FiniteReality — APPROVED** [review]
Seems good overall, just a few places where I'm not sure the code style is being properly followed.

**Comment by RubyNova** [comment]
This PR can just go. We can keep theb ranch around for ref.

**Pull request closed without merging**
